vmm/src/lib.rs

```diff
@@ -758,14 +783,17 @@ impl Vmm {
     ) -> Result<Self> {
         let mut epoll_context = EpollContext::new()?;
         // If this fails, it's fatal; using expect() to crash.
-        let api_event = epoll_context
-            .add_event(api_event_fd, EpollDispatch::VmmActionRequest)
+        epoll_context
+            .add_event(&api_event_fd, EpollDispatch::VmmActionRequest)
             .expect("Cannot add API eventfd to epoll.");
 
-        let write_metrics_event = epoll_context
+        let write_metrics_event_fd =
```

Comment:
Firstly, this has to do with the way I changed add_event. The old way in which add_event was written involved moving an EventFd (which was just a wrapper around an AsRawFd) into the add_event function, then moving it out again through the return value-- which was why it used to be necessary to write the code in the old way (you really needed to get the file descriptor from the return value again). This seemed convoluted, and looks like something that should be done via borrowing. I changed the function so that it receives a reference to the event file descriptor. But now, due to lifetimes, it is not possible to return the reference again.
Secondly,

Reply:
This is a rather big change. I would put it in a separate commit and explain in the commit message why you decided to make the change.

Reply:
Sure, I will do this